nautobot/extras/api/serializers.py

```diff
@@ -107,7 +110,29 @@
 #
 
 
-class NautobotModelSerializer(RelationshipModelSerializerMixin, CustomFieldModelSerializer, ValidatedModelSerializer):
+class NotesSerializerMixin(BaseModelSerializer):
+    """Extend Serializer with a `notes` field."""
+
+    notes = serializers.SerializerMethodField()
+
+    def get_field_names(self, declared_fields, info):
+        """Ensure that fields includes "notes" field if applicable."""
+        fields = list(super().get_field_names(declared_fields, info))
+        if hasattr(self.Meta.model, "notes"):
+            self.extend_field_names(fields, "notes")
+        return fields
+
+    @extend_schema_field(serializers.URLField())
+    def get_notes(self, instance):
+        notes_url = f"{instance._meta.app_label}-api:{instance._meta.model_name}-notes"
+        if instance._meta.app_label in settings.PLUGINS:
+            notes_url = f"plugins-api:{notes_url}"
+        return reverse(notes_url, args=[instance.id], request=self.context["request"])
+
+
+class NautobotModelSerializer(
+    RelationshipModelSerializerMixin, CustomFieldModelSerializer, ValidatedModelSerializer, NotesSerializerMixin
```

Comment:
Is this order needed for something to work right? Or could it be for consistency:

```suggestion
    RelationshipModelSerializerMixin, CustomFieldModelSerializer, NotesSerializerMixin, ValidatedModelSerializer
```